addons/hw_drivers/tools/helpers.py

```diff
@@ -248,7 +248,7 @@ def load_iot_handlers():
             if spec:
                 module = util.module_from_spec(spec)
                 spec.loader.exec_module(module)
-    http.addons_manifest = {}
+    module.addons_manifest = {} # to check
     http.root = http.Root()
```

Comment:
Since they were resetting the application, I would expect the intention was that the iot modules would be found and their controllers loaded (/cc @aprieels do you confirm? neither the docstring nor the commit message really explain why's this being done)

This is not going to do that, as the reset `http.root` doesn't load anything anymore.

Reply:
Since we load the manifest when loading, this line is no longer necessary if I understand correctly?

Reply:
The lines above *add* controllers to the set by executing python modules which (I assume) are not otherwise loaded.

Reply:
It used to be but it is no longer as the function is no longer in root.
https://github.com/odoo/odoo/pull/74758/files#diff-5f1b777eb0bc30f3f494ffe3e1522524cd0822f6c94c8644070a6db7c40b9c34L1314

I think `...addons_manifest = {}` should be deleted altogether.

Reply:
> It used to be but it is no longer as the function is no longer in root.

"It doesn't work anymore because I broke it" is not a compelling argument.

Reply:
I add `module.load_addons_manifest()` juste after the `module.addons_manifest = {}` so that the manifest is complete.

Reply:
Ok there are no more errors on my side

Reply:
I will merge all the commits into one and it will be good.

Reply:
Aaah right so the problem was not with the expression itself but with its side-effects in rebinding an other variable.

That should probably be explained in the commit message as it's a bit confusing / easy to miss (as demonstrated by me having completely missed it).

Reply:
@xmo-odoo I added a commit to have less change. And no longer have this less understandable part. What do you think ?